modules/tlc2/overrides/Json.java

```diff
@@ -345,114 +339,119 @@ private static JsonNode getArrayNode(IValue value) throws IOException {
-      case BOOLEAN:
-        return new BoolValue(node.asBoolean());
-      case STRING:
-        return new StringValue(node.asText());
-      case NULL:
-        return null;
-      default:
-        throw new IOException("Cannot convert value: unsupported JSON type " + node.getNodeType());
+  private static Value getValue(JsonElement node) throws IOException {
+    if (node.isJsonArray()) {
+      return getTupleValue(node);
+    }
+    else if (node.isJsonObject()) {
+      return getRecordValue(node);
     }
+    else if (node.isJsonPrimitive()) {
+      JsonPrimitive primitive = node.getAsJsonPrimitive();
+      if (primitive.isNumber()) {
+        return IntValue.gen(primitive.getAsInt());
+      }
+      else if (primitive.isBoolean()) {
+        return new BoolValue(primitive.getAsBoolean());
+      }
+      else if (primitive.isString()) {
+        return new StringValue(primitive.getAsString());
+      }
+    }
+    else if (node.isJsonNull()) {
+      return null;
+    }
+    throw new IOException("Cannot convert value: unsupported JSON value " + node.toString());
   }
 
   /**
-   * Converts the given {@code JsonNode} to a tuple.
+   * Converts the given {@code JsonElement} to a tuple.
    *
-   * @param node the {@code JsonNode} to convert
+   * @param node the {@code JsonElement} to convert
    * @return the tuple value
    */
-  private static TupleValue getTupleValue(JsonNode node) throws IOException {
+  private static TupleValue getTupleValue(JsonElement node) throws IOException {
     List<Value> values = new ArrayList<>();
-    for (int i = 0; i < node.size(); i++) {
-      values.add(getValue(node.get(i)));
+    JsonArray jsonArray = node.getAsJsonArray();
+    for (int i = 0; i < jsonArray.size(); i++) {
+      values.add(getValue(jsonArray.get(i)));
     }
     return new TupleValue(values.toArray(new Value[0]));
   }
 
   /**
-   * Converts the given {@code JsonNode} to a record.
+   * Converts the given {@code JsonElement} to a record.
    *
-   * @param node the {@code JsonNode} to convert
+   * @param node the {@code JsonElement} to convert
    * @return the record value
    */
-  private static RecordValue getRecordValue(JsonNode node) throws IOException {
+  private static RecordValue getRecordValue(JsonElement node) throws IOException {
     List<UniqueString> keys = new ArrayList<>();
     List<Value> values = new ArrayList<>();
-    Iterator<Map.Entry<String, JsonNode>> iterator = node.fields();
+    Iterator<Map.Entry<String, JsonElement>> iterator = node.getAsJsonObject().entrySet().iterator();
     while (iterator.hasNext()) {
-      Map.Entry<String, JsonNode> entry = iterator.next();
+      Map.Entry<String, JsonElement> entry = iterator.next();
       keys.add(UniqueString.uniqueStringOf(entry.getKey()));
       values.add(getValue(entry.getValue()));
     }
     return new RecordValue(keys.toArray(new UniqueString[0]), values.toArray(new Value[0]), false);
   }
 
-
```

Comment:
Please mark the `resolve` method and its counterpart in `TLCOverride` for removal after graduation to TLC.

Reply:
Added a @Deprecated annotation, thanks!